documentation/api/query/v4/pql.markdown

```diff
@@ -0,0 +1,361 @@
+
+    nodes[certname] { certname ~ '^web' }
+
+In this case, this would return only the certname field of nodes starting with `web`.
+
+## Entity
+
+The entity or context of a query (or subquery) defines what results you will get returned when
+performing a query, and provides the main context for any projections or filters in the query.
+There are many entities, for a full list see the [entities] documentation.
+
+For PQL queries, the entity context is the minimal amount of information one must provide, as it
+defines the results returned. For example, if you wanted to see all node information, you could
+provide a query as follows:
+
+    nodes {}
+
+And it would be enough to return all node data, without filtering or pagination.
+
+The entity context can also be used within a subquery, see the [subquery] section for more details.
+
+## Projection
+
+The projection part of a query, provides a mechanism to choose a subset of fields that are returned
+or to modify the way those fields are displayed with the usage of functions.
+
+The projection, lives within the brackets of a query:
+
+    nodes[<projection>] {}
+
+And is a comma seperated list of fields and functions:
+
+    facts[name, count()] { group by name }
+
+### Fields
+
+Entity field selection is an optional capability to ensure that only certain fields are returned in
+a response.
+
+For a basic query, if you don't provide any entity fields all data gets returned. However this can
+be be inefficient for both the database and the network to do this. By providing an entity field
+however, you can reduce what fields are returned.
+
+The entity field section of a query, can contain a number of field names separated by a comma:
+
+    entity[field1, field2, field3] {}
+
+As an example, to return only `certnames` for all nodes:
+
+    nodes[certnames] {}
+
+Or to return both the `name` and `value` of all facts:
+
+    facts[name, value] {}
+
+Only fields that are available for the entity type can be returned by PQL today.
+
+### Functions
+
+PQL supports the usage of some functions in the projection, but only aggregate functions are
+available today.
+
+Aggregate functions perform a calculation on a set of values and return a single value. Functions
+themselves are provided in the projection much like fields are:
+
+    entity[function(argument)] {}
+
+As an example, to query how many objects exist that start with a certname of `web` you could use
+the following filter and function combination:
+
+    nodes[count()] { certname ~ 'web.*' }
+
+There are only a few functions that are supported today by PQL, see the list below for more detail.
+
+#### `count()`
+
+Returns the number of objects returned by the query, instead of returning the actual results.
+
+#### `avg(<field>)`
+
+Returns the average value for the values held in the `<field>` argument.
+
+#### `sum(<field>)`
+
+Returns the sum of values for the values held in the `<field>` argument.
+
+#### `min(<field>)`
```

Comment:
This is a not-known issue, and pre-existing also for AST. I'm going to raise a bug on it.